### PR TITLE
Character interface added for easier implementation of future classes

### PR DESCRIPTION
**Pull request by petr7555**

### CODE REVIEW

**Comment by petr7555** [comment]
@AlexDvorak here it is ;)

File: src/main/java/com/crystal/ManFight.java [hunk 1]
```diff
@@ -71,34 +71,34 @@ public void startFight() {
...
-                return manAttacking.roar(manBeingAttacked);
+                return manAttacking.roar(characterBeingAttacked);
             default:
                 System.out.println(move + " is not a valid move you skip your turn doofus!");
                 return true;
         }
     }
 
-    public boolean doMoveAsMagician(String move, Man manAttacking) {
+    public boolean doMoveAsMagician(String move, Magician manAttacking) {
```
Comment (AlexDvorak) [inline src/main/java/com/crystal/ManFight.java]: I'm not a huge fan of this, I would rather keep this logic inside the individual character classes, maybe have a an attack method?

File: src/main/java/com/crystal/Monster.java [hunk 2]
```diff
@@ -26,4 +27,55 @@ public void rankMonster() {
...
+
+    @Override
+    public boolean headbutt(Character characterBeingAttacked) {
+        return false;
+    }
+
+    @Override
+    public boolean roar(Character characterBeingAttacked) {
+        return false;
+    }
```
Comment (AlexDvorak) [inline src/main/java/com/crystal/Monster.java]: Can we make these private and have the attack method call them?

File: src/main/java/com/crystal/Monster.java [hunk 3]
```diff
@@ -26,4 +27,55 @@ public void rankMonster() {
...
+
+    @Override
+    public void setScared(boolean scared) {
+        this.scared = scared;
+    }
+
+    @Override
+    public void isInjured(String attack) {
+
+    }
```
Comment (AlexDvorak) [inline src/main/java/com/crystal/Monster.java]: this should return a boolean given the name? even if it just returns true as a default for now

File: src/main/java/com/crystal/ManFight.java (hunk 1 above)
Comment (petr7555) [reply src/main/java/com/crystal/ManFight.java]: done

File: src/main/java/com/crystal/Monster.java (hunk 2 above)
Comment (petr7555) [reply src/main/java/com/crystal/Monster.java]: done

File: src/main/java/com/crystal/Monster.java (hunk 3 above)
Comment (petr7555) [reply src/main/java/com/crystal/Monster.java]: renamed to be unambiguous

**Comment by Vertfromage** [comment]
Looks good to me, thanks petr7555, and thanks AlexDvorak!

**Pull request merged**